components/eam/src/physics/cam/phys_grid.F90

```diff
@@ -1098,7 +1098,9 @@ subroutine phys_grid_init( )
              if (iam == owner_d) then
                 if (.not. associated(btofc_blk_offset(blockids(jb))%pter)) then
                    blksiz = get_block_gcol_cnt_d(blockids(jb))
+!DIR$ NOINLINE
```

Comment:
You probably need to add an ifdef guard as this issue is specific to Crusher and Cray compiler IIRC.
This would be an interim fix until the Cray compiler fix arrives in a newer release.